cmd/crc/cmd/console.go

```diff
@@ -48,6 +48,12 @@ func runConsole(arguments []string) {
 	consoleConfig := machine.ConsoleConfig{
 		Name: constants.DefaultName,
 	}
+
+	// Check if the crc instance is exist
+	if _, _, err := machine.GetHost(consoleConfig.Name); err != nil {
+		errors.ExitWithMessage(1, err.Error())
+	}
+
```

Comment:
I don't think all these tests should be done in cmd/crc/cmd/, they belong in the corresponding pkg/crc/machine/ methods so that the system tray gets the same behaviour. Only "crc delete" is a bit trickier if you want to avoid the user interaction when the VM is not running.

Reply:
It's also missing a `defer firstRetVal.Close()` similar to what you fixed in the previous commit ;)

Reply:
@cfergeau right so system tray also should first try to see the exist of the VM and the only run other operation `GetHost` is something done in the `pkg/crc/machine` side not part of the `cmd` side. 

> It's also missing a defer firstRetVal.Close() similar to what you fixed in the previous commit ;)

Ah right, will do that.

Reply:
> It's also missing a defer firstRetVal.Close() similar to what you fixed in the previous commit ;)

Done, good catch 👍

Reply:
> @cfergeau right so system tray also should first try to see the exist of the VM and the only run other operation 

My understanding is that this is precisely the kind of code duplication we want to avoid "in order to implement 'Stop', the systray implementation should do X, then Y, then Z". It's probably fine to have some sort of generic "preflight" check which has to be called before calling "Stop()"

Reply:
> probably fine to have some sort of generic "preflight" check which has to be called before calling "Stop()"

What do you mean by having a kind of `preflight` , doesn't `GetHost` is actually doing it? Also on the tray side if you want to just consume the `Stop` , `Start` you can directly do since those functionality doesn't change at all.